bookshelf/src/DataModel/MongoDb.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+namespace Google\Cloud\Samples\Bookshelf\DataModel;
+
+use \MongoDB\BSON\ObjectId;
+
+/**
+ * Class MongoDb implements the DataModel with MongoDB.
+ *
+ * To use this class, set two environment variables; MONGO_URL and
+ * MONGO_NAMESPACE.
+ */
+class MongoDb implements DataModelInterface
+{
+    /**
+     * Properties of the books.
+     *
+     * @var array
+     */
+    protected $columns = array(
+        'id',
+        'title',
+        'author',
+        'publishedDate',
+        'imageUrl',
+        'description',
+        'createdBy',
+        'createdById',
+    );
+
+    /**
+     * MongoDB collection.
+     *
+     * @var \MongoDB\Collection
+     */
+    private $db;
+
+    /**
+     * Connects to the MongoDB server.
+     */
+    public function __construct()
+    {
+        if (false == ($dbUrl = getenv('MONGO_URL'))) {
+            throw new \Exception(
+                'Set the environment variable MONGO_URL to the URL of your '
+                . 'MongoDB instance.');
+        }
+        if (false == ($namespace = getenv('MONGO_NAMESPACE'))) {
+            throw new \Exception(
+                'Set the environment variable MONGO_NAMESPACE to your '
+                . 'MongoDB namespace (in the "db.collection" format).');
+        }
+        $manager = new \MongoDB\Driver\Manager($dbUrl);
+        $this->db = new \MongoDB\Collection($manager, $namespace);
+    }
+
+    public function listBooks($limit = 10, $cursor = null)
+    {
+        if ($cursor) {
+            $q = $this->db->find(
+                array('_id' => array('$gt' => new ObjectId($cursor))),
+                array('sort' => array('_id' => 1))
+            );
+        } else {
+            $q = $this->db->find(
+                array(),
+                array('sort' => array('_id' => 1))
+            );
+        }
+        $rows = array();
+        $last_row = null;
+        $new_cursor = null;
+        foreach ($q as $row) {
+            if (count($rows) == $limit) {
+                $new_cursor = (string) ($last_row->_id);
+                break;
+            }
+            array_push($rows, $this->bookToArray($row));
+            $last_row = $row;
+        }
+        return array(
+            'books' => $rows,
+            'cursor' => $new_cursor,
+        );
+    }
+
+    public function create($book, $id = null)
```

Comment:
we *could* add a docblock with `@see DataModelInterface::listBooks` for those not using IDEs